src/calliope/backend/expression_parser.py

```diff
@@ -240,7 +241,7 @@ def _operator_operands(
 
     def _apply_where_array(self, evaluated: xr.DataArray) -> xr.DataArray:
         """Util function to apply where arrays to non-latex strings."""
-        where_array = self.eval_attrs["where_array"]
+        where_array = self.eval_attrs.get("where_array", xr.DataArray(True))
```

Comment:
Now we have `where` parsers using this evaluator but we don't define `where_array` for them (they are busy trying to generate it!), so this is a default for it. An alternative would be to set that as the value for all `where` parsers to have in their `eval_attrs` so that we can revert to `self.eval_attrs["where_array"]` here.